### PR TITLE
Writing a primer/utilities plugin to look for code that duplicates utilities

### PR DESCRIPTION
**Pull request by jonrohan**
Checks for selectors with single declarations that can be replaced with [primer/css utilities](https://primer.style/css/utilities/).
```scss
.foo {
  color: var(--color-fg-default);
}
/**          ↑
 *           FAIL: --color-fg-default can be replaced with .color-fg-default */

.foo {
  color: #custom;
}
/**          ↑
 *           OK: Color value doesn't match a utility. */

.foo {
  color: var(--color-fg-default);
  padding: 0;
}
/**          ↑
 *           OK: Has more than one declaration, not considered */
```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -69,6 +70,7 @@ module.exports = {
     'primer/responsive-widths': true,
     'primer/spacing': true,
     'primer/typography': true,
+    'primer/utilities': null,
```
Comment (jonrohan) [inline index.js:73]: I turned this off by default as it's pretty specific circumstances you'd want to turn it on

File: plugins/utilities.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+const stylelint = require('stylelint')
+const utilities = require('./lib/primer-utilities')
```
Comment (jonrohan) [inline plugins/utilities.js:2]: For now this is a lib file, but I envision pulling this in from primer/css or primer/primitives

**Review by simurai — APPROVED** [review]
This makes sense to me. 👍 

> Checks for selectors with single declarations

What about states or other pseudo selectors like

```scss
.foo:hover {
  color: var(--color-fg-default);
}

.foo:first-child {
  color: var(--color-fg-default);
}
```

I assume they would be OK too?

**Comment by jonrohan** [comment]
> What about states or other pseudo selectors like

I believe they're ok because of the class match regex I wrote, however it's a good test to add so we can be sure we don't regress someday 👍🏻

**Pull request merged**